### PR TITLE
Remove workflows, scripts and references to MBED OS in documentation

### PR DESCRIPTION
**Pull request by ewertons**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

azure-iot-sdk-c has not supported MBED OS for a while, and we currently have no running tests for that platform.
There are leftover code and doc references in azure-iot-sdk-c that should be removed since we do not officially support MBED OS.

# Description of the solution
 
Remove all references to MBED OS in docs, code and workflows.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -207,7 +207,7 @@ This folder contains application development guides and device setup instruction
 
 Contains Azure IoT Hub client components that provide the raw messaging capabilities of the library. Refer to the API documentation and samples for information on how to use it.
 
-   * build: has one subfolder for each platform (e.g. Windows, Linux, Mbed). Contains makefiles, batch files, and solutions that are used to generate the library binaries.
+   * build: has one subfolder for each platform (e.g. Windows, Linux). Contains makefiles, batch files, and solutions that are used to generate the library binaries.
```
Comment (ewertons) [inline readme.md:210]: Do not merge this PR until author gets offline approval from team.

File: readme.md [hunk 2]
```diff
@@ -217,7 +217,7 @@ This folder contains application development guides and device setup instruction
 
 Contains Azure IoT Hub client components that provide the raw messaging capabilities of the library. Refer to the API documentation and samples for information on how to use it.
 
-   * build: has one subfolder for each platform (e.g. Windows, Linux, Mbed). Contains makefiles, batch files, and solutions that are used to generate the library binaries.
```
Comment (jspaith) [inline readme.md:220]: Nits!  Some additional floating "mbed" references.

There are more mbed build directories

```
iothub_client\build\iothub_amqp_transport\mbed
iothub_client\build\iothub_http_transport\mbed
iothub_client\build\iothub_mqtt_transport\mbed
iothub_client_sample_upload_to_blob\mbed
iothub_client\samples\iothub_client_sample_upload_to_blob_mb\mbed
iothub_client\samples\iothub_ll_c2d_sample\mbed
samples\iothub_ll_client_x509_sample\mbed
samples\iothub_ll_telemetry_sample\mbed
serializer\build\mbed
doc\media\mbed1.* # This seems to be not linked by any docs but you should double check
```

* In `serializer\readme.md` please remove the "mbed" reference.

```
you will find detailed instructions on how to use on Linux, mbed, Windows and other platforms
```

In `iothub_client\samples\iothub_client_device_twin_and_methods_sample\iothub_client_device_twin_and_methods_sample.c` 

```c
// For mbed add the certificate information
```

Recomend replacing this with our telemetry sample notes on trusted certificates
```c
        // Setting the Trusted Certificate. This is only necessary on systems without
        // built in certificate stores.
```

We have `MBED_BUILD_TIMESTAMP` #ifdef's floating around in samples and long haul tests I'm assuming are MBED OS and not MBED TLS related but you should double check me.

**Review by jspaith — CHANGES_REQUESTED** [review]
Requesting changes per my comment inline - what you're deleting looks good but there's some additional to remove.

**Review by CIPop — APPROVED** [review]
Thanks @ewertons !

**Review by danewalton — APPROVED** [review]

File: readme.md (hunk 2 above)
Comment (ewertons) [reply readme.md:220]: I am really surprised I missed those. Wow. Thanks for the list and sorry for missing them.

MBED_BUILD_TIMESTAMP is related to MBED OS, not mbed TLS, and should be removed as well. 
https://github.com/ARMmbed/mbed-os/issues/10228

Comment (jspaith) [reply readme.md:220]: Also note around line 70 another stray mbed reference - I think you can safely remove the ",mbed".
```
Socket                    | [socketio\_\*](https://github.com/Azure/azure-c-shared-utility/tree/master/adapters)                                | Provides a wrapper over the specific socket API ([win32, berkeley, mbed](https://github.com/Azure/azure-c-shared-utility/tree/master/adapters)), using the [xio](https://github.com/Azure/azure-c-shared-utility/blob/master/inc/azure_c_shared_utility/xio.h) interface that the device client SDK uses
```

**Review by jspaith — APPROVED** [review]
I added one more nit in comments but otherwise looks good!  Thank you

File: readme.md (hunk 2 above)
Comment (ewertons) [reply readme.md:220]: Fixed. Thanks!

**Pull request merged**